packages/core/upload/admin/src/pages/App/ConfigureTheView/index.js

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useReducer } from 'react';
+import { useIntl } from 'react-intl';
+import isEqual from 'lodash/isEqual';
+import PropTypes from 'prop-types';
+
+import { Layout, HeaderLayout, ContentLayout } from '@strapi/design-system/Layout';
+import {
+  useNotification,
+  ConfirmDialog,
+  Link,
+  useTracking,
+  useFocusWhenNavigate,
+} from '@strapi/helper-plugin';
+import Check from '@strapi/icons/Check';
+import { Button } from '@strapi/design-system/Button';
+import ArrowLeft from '@strapi/icons/ArrowLeft';
+import { Main } from '@strapi/design-system/Main';
+import { Settings } from './components/Settings';
+
+import reducer, { initialState } from './state/reducer';
+import init from './state/init';
+import { onChange, setLoaded } from './state/actions';
+
+import pluginID from '../../../pluginId';
+import getTrad from '../../../utils/getTrad';
+import { useConfig } from '../../../hooks/useConfig';
+
+const ConfigureTheView = ({ config }) => {
+  const { trackUsage } = useTracking();
+  const { formatMessage } = useIntl();
+  const toggleNotification = useNotification();
+  const { mutation } = useConfig();
+  const { isLoading: isSubmittingForm } = mutation;
+
+  const [showWarningSubmit, setWarningSubmit] = useState(false);
+  const toggleWarningSubmit = () => setWarningSubmit((prevState) => !prevState);
+
+  const [reducerState, dispatch] = useReducer(reducer, initialState, () => init(config.data));
```

Comment:
potentially loaded question, is a reducer a bit overkill for this? It's two actions and some modified vs initial state 🤔 

cc @gu-stav

Reply:
I don't have strong opinion tbh. How was it done for Configure The View in the CM? What do you think would be the benefit to replace this with `useState()` (I guess that is what you had in mind)?

Reply:
I think considering how small the reducer is it just removes the overhead of having an additional 3 files and creates transparency when reading the code.

Reply:
Alright, thanks! As I said: I don't have a strong opinion here and am okay with both solutions. I know we should solve only our current problems, but given that the CTV in the CM already has even more ways to configure the view (customizing column names ...) I think there is a good chance we will add them here, too and then it could help to have this setup already done ...

Reply:
Okay great, if thats how other parts are working lets do that. I do think we need a wider discussion about `useReducer` vs `Context` vs `Redux` and which one we all believe to be using, its getting wild 😓